phpunit/navigation-block-fallback-test.php

```diff
@@ -0,0 +1,320 @@
+<?php
+/**
+ * Tests Fallback Behavior for Navigation block
+ *
+ * @package WordPress
+ */
+
+class Tests_Block_Navigation_Fallbacks extends WP_UnitTestCase {
+
+	public function set_up() {
+		parent::set_up();
+		// Navigation menu will be created on Theme switch. Therefore in order to test
+		// the behaviour of `gutenberg_block_core_navigation_create_fallback`
+		// the auto-creation of a fallback must be disabled for this initial
+		// theme switch.
+		add_filter( 'block_core_navigation_skip_fallback', '__return_true' );
+		switch_theme( 'emptytheme' );
+		remove_filter( 'block_core_navigation_skip_fallback', '__return_true' );
+	}
+
+	private function get_navigations_in_database() {
+		$navs_in_db = new WP_Query(
+			array(
+				'post_type'      => 'wp_navigation',
+				'post_status'    => 'publish',
+				'posts_per_page' => -1,
+				'orderby'        => 'date',
+				'order'          => 'DESC',
+			)
+		);
+
+		return $navs_in_db->posts ? $navs_in_db->posts : array();
+	}
+
+	private function mock_wp_install() {
+		$user = get_current_user();
+		do_action( 'wp_install', $user );
```

Comment:
I checked with @anton-vlasenko and @ironprogrammer about this and they felt confidence it would be safe to trigger on a mock basis here.

Reply:
Ah...looks like they don't which is what I understand you are alluding to.

Question: do we _need_ to pass the User argument? If so would you be able to point me in the direction of a test which does this setup step?

Much appreciated.

Reply:
I looked again at https://developer.wordpress.org/reference/functions/wp_get_current_user/.

It says:

> Will set the current user, if the current user is not set...If no user is logged-in, then it will set the current user to 0, which is invalid and won’t have any permissions.

AKAIK our tests have no dependency on the user object so either:

- We cannot remove the `get_current_user()` call
- Allow it to run and set the current user to `0`.

If that's incorrect I'd be grateful if you could explain the best practice so I'm aware for future reference.